resolwe/flow/views.py

```diff
@@ -135,23 +136,51 @@ class ResolweCreateDataModelMixin(ResolweCreateModelMixin):
     """
     def create(self, request, *args, **kwargs):
         collections = request.data.get('collections', [])
-        if len(collections) != 1:
-            return Response({'collections': 'Exactly one id required on create.'},
-                            status=status.HTTP_400_BAD_REQUEST)
 
-        try:
-            collection = Collection.objects.get(pk=collections[0])
-        except Collection.DoesNotExist:
-            return Response({'collections': ['Invalid pk "{}" - object does not exist.'.format(collections[0])]},
+        for collection_id in collections:
+            try:
+                collection = Collection.objects.get(pk=collection_id)
+            except Collection.DoesNotExist:
+                return Response({'collections': ['Invalid pk "{}" - object does not exist.'.format(collection_id)]},
+                                status=status.HTTP_400_BAD_REQUEST)
+
+            if not request.user.has_perm('add_collection', obj=collection):
+                if request.user.is_authenticated():
+                    raise exceptions.PermissionDenied
+                else:
+                    raise exceptions.NotFound
+
+        process_slug = request.data.get('process', None)
+        process_quey = Process.objects.filter(slug=process_slug).order_by('version')
+        if not process_quey.exists():
+            # XXX: security - is it ok to reveal which processes (don't) exist?
+            return Response({'process': ['Invalid process slug "{}" - object does not exist.'.format(process_slug)]},
                             status=status.HTTP_400_BAD_REQUEST)
+        process = process_quey.last()
+        request.data['process'] = process.pk
 
-        if not request.user.has_perm('add_collection', obj=collection):
+        if not request.user.has_perm('view_process', obj=process):
             if request.user.is_authenticated():
                 raise exceptions.PermissionDenied
             else:
                 raise exceptions.NotFound
 
-        return super(ResolweCreateDataModelMixin, self).create(request, *args, **kwargs)
+        resp = super(ResolweCreateDataModelMixin, self).create(request, *args, **kwargs)
+        manager.communicate()
```

Comment:
At the moment you get `500 error` (without traceback if `DEBUG` is `False`). I think that `500` is appropriate error, but we can add helpful message. I'll add task for this and do it later, because it is not crucial for the moment.